src/model/ai/evaluation/PawnDistanceToKingedEvaluator.java

```diff
@@ -0,0 +1,54 @@
+package model.ai.evaluation;
+
+import model.Board;
+import model.PieceColor;
+import model.PieceInterface;
+import model.Square;
+
+public class PawnDistanceToKingedEvaluator implements BoardEvaluatorInterface {
+
+    private static BoardEvaluatorInterface instance = null;
+
+    public static BoardEvaluatorInterface getInstance() {
+        if (instance == null) {
+            return new PawnDistanceToKingedEvaluator();
+        }
+        return instance;
+    }
+
+    private PawnDistanceToKingedEvaluator() {
+    }
+
+    @Override
+    public double evaluateBoard(Board theBoard, PieceColor color) {
+        double score = 0.0;
+
+        for (final Square square : theBoard.getGameState()) {
+
+            if (square.isOccupied()) {
+                final PieceInterface piece = square.getOccupyingPiece();
+
+                if (!piece.isKing()) {
+                    final int rowNumber = square.getRowNumber();
+                    int distanceToOtherSide = 0;
+
+                    if (piece.getColor() == PieceColor.BLACK) {
+                        distanceToOtherSide = 8 - rowNumber;
+                    } else {
+                        distanceToOtherSide = rowNumber - 1;
+                    }
+
+                    final double value = 8 - distanceToOtherSide;
+                    if (piece.getColor() == color) {
+                        score += value;
+                    } else {
+                        score -= value;
```

Comment:
I'm undecided, but do you think this evaluator should definitely take into account opponents positions, or just it's own? Also, for our searching, are we passing the opposite color into the evaluators when it's not our turn? Just wanted to make sure of that.